### PR TITLE
polish(rjy): polish pg/iqn/edac policy doc

### PR DESCRIPTION
**Pull request by nighood**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/policy/iqn.py [hunk 1]
```diff
@@ -135,6 +159,34 @@ def _forward_learn(self, data: dict) -> Dict[str, Any]:
         Returns:
```
Comment (HarryXuancy) [inline ding/policy/iqn.py:159]: delete the old comment

File: ding/policy/iqn.py [hunk 2]
```diff
@@ -186,13 +238,30 @@ def _forward_learn(self, data: dict) -> Dict[str, Any]:
...
         }
 
     def _load_state_dict_learn(self, state_dict: Dict[str, Any]) -> None:
+        """
+        Overview:
+            Load the state_dict variable into policy learn mode.
+        Arguments:
+            - state_dict (:obj:`Dict[str, Any]`): The dict of policy learn state saved before.
+
+        .. tip::
```
Comment (HarryXuancy) [inline ding/policy/iqn.py:260]: what is the difference between tip and note

File: ding/policy/pg.py [hunk 3]
```diff
@@ -76,13 +97,31 @@ def _init_learn(self) -> None:
...
         Arguments:
-            - data (:obj:`dict`): Dict type data, including at least ['obs', 'action', 'reward', 'next_obs','adv']
+            - data (:obj:`List[Dict[int, Any]]`): The input data used for policy forward, including the latest \
+                collected training samples for on-policy algorithms like PG. For each element in list, the key of the \
+                dict is the name of data items and the value is the corresponding data. Usually, the value is \
+                torch.Tensor or np.ndarray or there dict/list combinations. In the ``_forward_learn`` method, data \
+                often need to first be stacked in the batch dimension by some utility functions such as \
+                ``default_preprocess_learn``. \
+                For PG, each element in list is a dict containing at least the following keys: ``obs``, ``action``, \
+                ``return``.
```
Comment (HarryXuancy) [inline ding/policy/pg.py:113]: is this reward or return?and why the elements reduce from 5 to 3

File: ding/policy/edac.py [hunk 4]
```diff
@@ -151,6 +164,34 @@ def _init_learn(self) -> None:
...
+                detailed definition of the dict, refer to the code of ``_monitor_vars_learn`` method.
+
+        .. note::
+            The input value can be torch.Tensor or dict/list combinations and current policy supports all of them. \
+            For the data type that not supported, the main reason is that the corresponding model does not support it. \
+            You can implement you own model rather than use the default model. For more information, please raise an \
+            issue in GitHub repo and we will continue to follow up.
+
+        .. note::
+            For more detailed examples, please refer to our unittest for DiscreteEDACPolicy: \
```
Comment (PaParaZz1) [inline ding/policy/edac.py:192]: There is no DiscreteEDAC

File: ding/policy/iqn.py [hunk 5]
```diff
@@ -13,9 +13,9 @@
 
 @POLICY_REGISTRY.register('iqn')
 class IQNPolicy(DQNPolicy):
-    r"""
+    """
     Overview:
-        Policy class of IQN algorithm.
+        Policy class of IQN algorithm. Paper link: https://arxiv.org/pdf/1806.06923.pdf.
```
Comment (PaParaZz1) [inline ding/policy/iqn.py:18]: add some information about its related area (distrbutional RL)

File: ding/policy/pg.py [hunk 6]
```diff
@@ -76,13 +97,31 @@ def _init_learn(self) -> None:
         self._learn_model = self._model  # for compatibility
 
     def _forward_learn(self, data: dict) -> Dict[str, Any]:
```
Comment (PaParaZz1) [inline ding/policy/pg.py:99]: check all the typing lint and make them follow our formats.

**Review by PaParaZz1 — APPROVED** [review]
